src/3d/qgsrubberband3d.h

```diff
@@ -122,6 +143,14 @@ class _3D_EXPORT QgsRubberBand3D
 
     void setPoints( const QgsLineString &points );
 
+    /**
+     * Sets the polygon geometry to be shown
+     *
+     * \note Function doesn't set \a GeometryType to Polygon
+     * \since QGIS 3.44
+     */
+    void setPolygon( const QgsPolygon &polygon );
```

Comment:
it would be nicer to actually have a more generic approach - e.g. `setGeometry( const QgsGeometry &geometry );` - so that in the future we can also support other geometry types ... and the API would be similar to QgsRubberBand used in 2D canvas